src/main.cpp

```diff
@@ -292,12 +292,8 @@ Napi::Value DDWAFContext::run(const Napi::CallbackInfo& info) {
   }
   if (code == DDWAF_MATCH) {
     res.Set("status", Napi::String::New(env, "match"));
-    res.Set("data", Napi::String::New(env, result.data));
-    Napi::Array actions = Napi::Array::New(env, result.actions.size);
-    for (uint32_t i = 0; i < result.actions.size; ++i) {
-      actions[i] = Napi::String::New(env, result.actions.array[i]);
-    }
-    res.Set("actions", actions);
+    res.Set("events", from_ddwaf_object(&result.events, env));
+    res.Set("actions", from_ddwaf_object(&result.actions, env));
```

Comment:
Are there `events` and `actions` only when there has been a match?

Reply:
Yes confirmed by Anil